### PR TITLE
feat: [PL-29028]: added option to resend 2fa email on users page

### PR DESCRIPTION
**Pull request by tiwaritejas**
### Summary

We will support the option to resend 2fa email to users from the users page in NG UI.

#### Screenshots

Screenshot 2023-02-20 at 2 52 18 PM
Screenshot 2023-02-20 at 2 45 54 PM
Screenshot 2023-02-15 at 6 19 48 PM

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by tiwaritejas** [comment]
retrigger sonar

**Comment by mayankVermaHarness** [comment]
The `icon` in first screenshot is not visible. Can you see if SVG has embedded colour or uses "currentColor"?

If it uses "currentColor", we can add color to parent of SVG and SVG will inherit it.

**Comment by abhinavrastogi-harness** [comment]
Few issues I see:
1. icon is missing in the menu
2. in the confirmation dialog, the word "user" is getting split into two lines. 
3. capitalisation on the string "Email New Two Factor Auth Secret" looks wrong. I think it should be "Email new Two Factor Auth secret"

**Comment by tiwaritejas** [comment]
retrigger coverage

**Comment by tiwaritejas** [comment]
retrigger sonar

**Comment by tiwaritejas** [comment]
retrigger cypress-cv

**Comment by tiwaritejas** [comment]
retrigger titlecheck

**Comment by tiwaritejas** [comment]
retrigger jest

**Review by tiwaritejas — COMMENTED** [review]
@abhinavrastogi-harness @mayankVermaHarness Addressed your feedback. Please see the updated screenshots. Thanks!

File: src/modules/20-rbac/pages/Users/views/ActiveUsersListView.tsx [hunk 1]
```diff
@@ -213,6 +214,33 @@ const RenderColumnMenu: Renderer<CellProps<UserAggregate>> = ({ row, column }) =
...
+      if (didConfirm && data) {
+        try {
+          resetTwoFactorAuthPromise({
+            queryParams: {
+              accountIdentifier: accountId
+            },
+            userId: data.uuid
+          })
+          const sent = true
+          sent && showSuccess(getString('rbac.usersPage.resendTwoFactorEmailSuccess', { name }))
```
Comment (mayankVermaHarness) [inline src/modules/20-rbac/pages/Users/views/ActiveUsersListView.tsx:235]: This `sent` variable currently is doing nothing.
You can `await` on `resetTWoFactorAuthPromise` and further showSuccess on the basis of response.

```
try {
        const response = await resetTwoFactorAuthPromise({
            queryParams: {
              accountIdentifier: accountId
            },
            userId: data.uuid
         })
         if (response.ok) {
             showSuccess(getString('rbac.usersPage.resendTwoFactorEmailSuccess', { name }))
         }
```

File: src/modules/20-rbac/pages/Users/views/ActiveUsersListView.tsx [hunk 2]
```diff
@@ -254,6 +282,10 @@ const RenderColumnMenu: Renderer<CellProps<UserAggregate>> = ({ row, column }) =
     }
   }
 
+  const resendTwoFactorEmail = async (): Promise<void> => {
```
Comment (mayankVermaHarness) [inline src/modules/20-rbac/pages/Users/views/ActiveUsersListView.tsx:285]:
```suggestion
  const resendTwoFactorEmail = (): void => {
```

**Comment by tiwaritejas** [comment]
retrigger cypress-pipeline

**Comment by tiwaritejas** [comment]
retrigger coverage

**Review by mayankVermaHarness — APPROVED** [review]

**Review by swarajcheguri-harness — APPROVED** [review]

**Review by abhinavrastogi-harness — APPROVED** [review]

**Pull request merged**